frontend/src/pages/Battle/Battle.tsx

```diff
@@ -11,11 +11,56 @@ import { Button } from "components/Button";
 import { useUnit } from "effector-react";
 import { logsStore } from "model/logs";
 import { battle } from "model/battleLogs";
-// import { userStore } from "model/user";
-// import { isEmpty } from "lodash";
-// import { useAlert } from "@gear-js/react-hooks";
 
 export type BattleProps = {};
+
+type TurnTypes = "move" | "attack" | "miss" | "rest";
+type TurnGeneric<K extends TurnTypes, T> = {
+  [key in K]: T;
+};
+type Turn =
+  | TurnGeneric<"move", { position: number }>
+  | TurnGeneric<"attack", { damage: number; position: number }>
+  | TurnGeneric<"miss", { position: number }>
+  | TurnGeneric<"rest", { position: number }>;
+
+type BattleLogs = {
+  logs: Array<
+    {
+      [key in `c${"1" | "2" | "3" | "4"}`]: string;
```

Comment:
`BattleLog` represents a battle between two gladiators so only c1 and c2 are possible values

Reply:
fixed